tests/end-to-end/external-solution-integration/pkg/step/step.go

```diff
@@ -23,14 +23,28 @@ type Runner struct {
 	cleanup CleanupMode
 }
 
+type RunnerOption func(runner *Runner)
+
+func WithCleanupDefault(mode CleanupMode) RunnerOption {
```

Comment:
(nit) I think the name `WithCleanup` is sufficient.

Reply:
as it can be overridden by a commandline argument this will not be the definitive final cleanup mode, it will be the default for this testcase